client/lib/js/utils/cryptoconditions/filter.js

```diff
@@ -0,0 +1,24 @@
+
+const filterByType = ({ condition, typeId, maxDepth }) => {
+    let res = [];
+    if (condition.hasOwnProperty('type_id') && condition.type_id === typeId) {
+        res.push(condition);
+    }
+
+    const nextMaxDepth = (maxDepth === parseInt(maxDepth, 10)) ? maxDepth - 1 : null;
```

Comment:
not sure

if `maxDepth === null || maxDepth === undefined` it should always execute recursively